src/services/services.ts

```diff
@@ -1873,20 +1870,28 @@ module ts {
         return createLanguageServiceSourceFile(sourceFile.fileName, scriptSnapshot, sourceFile.languageVersion, version, /*setNodeParents:*/ true);
     }
 
-    export function createDocumentRegistry(): DocumentRegistry {
+    function createGetCanonicalFileName(useCaseSensitivefileNames: boolean): (fileName: string) => string {

```

Comment:
this looks repeated, yet different.. is there a way to simplify this whole thing without a host.. if we can ask the host about whether to use casesenseitivity all the time instead of the function.. this would be a breaking API change though.